CDP4Common.Tests/Comparers/ActualFiniteStateComparerTestFixture.cs

```diff
@@ -0,0 +1,132 @@
+            this.possibleFiniteStateList_2.PossibleState.Add(this.possibleFiniteState_2_1);
+            this.possibleFiniteStateList_2.PossibleState.Add(this.possibleFiniteState_2_2);
+
+            this.actualFiniteStateList = new ActualFiniteStateList(Guid.NewGuid(), this.cache, this.uri);
+            this.actualFiniteStateList.PossibleFiniteStateList.Insert(1, this.possibleFiniteStateList_1);
+            this.actualFiniteState_1 = new ActualFiniteState(Guid.NewGuid(), this.cache, this.uri);
+            this.actualFiniteState_1.PossibleState.Add(this.possibleFiniteState_1_1);
+            this.actualFiniteState_2 = new ActualFiniteState(Guid.NewGuid(), this.cache, this.uri);
+            this.actualFiniteState_2.PossibleState.Add(this.possibleFiniteState_1_2);
+            this.actualFiniteState_3 = new ActualFiniteState(Guid.NewGuid(), this.cache, this.uri);
+            this.actualFiniteState_3.PossibleState.Add(this.possibleFiniteState_2_1);
+            this.actualFiniteState_4 = new ActualFiniteState(Guid.NewGuid(), this.cache, this.uri);
+            this.actualFiniteState_4.PossibleState.Add(this.possibleFiniteState_2_2);
+            this.actualFiniteStateList.ActualState.Add(this.actualFiniteState_1);
+            this.actualFiniteStateList.ActualState.Add(this.actualFiniteState_2);
+            this.actualFiniteStateList.ActualState.Add(this.actualFiniteState_3);
+            this.actualFiniteStateList.ActualState.Add(this.actualFiniteState_4);
+
+            this.comparer = new ActualFiniteStateComparer();
+        }
+
+        [Test]
+        public void Verify_that_when_containers_are_not_the_same_exception_is_thrown()
+        {
+            var otherActualFiniteStateList = new ActualFiniteStateList(Guid.NewGuid(), this.cache, this.uri);
+            var otherActualFiniteState = new ActualFiniteState(Guid.NewGuid(), this.cache, this.uri);
+
+            otherActualFiniteStateList.ActualState.Add(otherActualFiniteState);
+
+            Assert.Throws<InvalidOperationException>(() => this.comparer.Compare(this.actualFiniteState_1, otherActualFiniteState));
+        }
+
+        [Test]
+        public void Verify_that_when_states_are_the_same_result_is_zero()
+        {
+            var result = this.comparer.Compare(this.actualFiniteState_1, this.actualFiniteState_1);
+            Assert.That(0, Is.EqualTo(result));
+        }
+
+        [Test]
+        public void Verify_that_when_actualstates_contain_no_possiblestates_zero_is_returned()
+        {
+            this.actualFiniteState_1.PossibleState.Clear();
+            this.actualFiniteState_2.PossibleState.Clear();
+
+            var result = this.comparer.Compare(this.actualFiniteState_1, this.actualFiniteState_2);
+
+            Assert.That(0, Is.EqualTo(result));
+        }
+
+        [Test]
+        public void Verify_that_when_actualstate1_is_smaller_than_actualustate2_a_negative_number_is_returned()

```

Comment:
There is a small typo in the name Verify_that_when_actualstate1_is_smaller_than_actual**_u_**state2_a_negative_number_is_returned